src/item_factory.cpp

```diff
@@ -1199,20 +1199,24 @@ bool Item_factory::check_ammo_type( std::string &msg, const ammotype &ammo ) con
 
 void Item_factory::check_definitions() const
 {
+    auto is_container = []( const itype * t ) {
```

Comment:
```suggestion
    bool is_container = []( const itype * t ) {
```

Reply:
If only lambda expressions were so simple. It won't compile with the return type set to bool.

Auto return types are generally frowned upon, but are greatly simplify the code when used with lamdas. For proof open up git bash and change to src directory. Enter the following command: grep -rn "auto.*= \["

Reply:
Sigh... ah, well.